catalog/resource_catalog.go

```diff
@@ -75,6 +76,26 @@ func ResourceCatalog() *schema.Resource {
 				return err
 			}
 
+			// Bind the current workspace if the catalog is isolated, otherwise the read will fail
+			if d.Get("isolation_mode") == "ISOLATED" {
```

Comment:
reduce the nesting by flipping the if statement - e.g. if it's not isolated catalog, return immediately